### PR TITLE
NO-ISSUE: cloud.md needs a little refresh

### PR DESCRIPTION
**Pull request by omertuc**
It was using v1 instead of v2, and also had some typos / disorganization.

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [x] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @romfreiman

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: docs/cloud.md [hunk 1]
```diff
@@ -2,29 +2,29 @@
...
 Some APIs accept both types. See configuration in [API definition](../swagger.yaml).
 
 Agent authentication is used by the processes running on the nodes (Agent, Installer & Controller).
 The Agent APIs are not meant to be used by end users.
 
-#### User Authentication
+## User Authentication
 
-User Authentication is using JWT tokens. There are two ways to get the token:
+User Authentication is using JWT tokens. The JWT token is valid for 15 minutes and need to be provided in the header of the HTTP request. There are two ways to get the token:
```
Comment (carbonin) [inline docs/cloud.md:25]:
```suggestion
User Authentication is using JWT tokens. The JWT token is valid for 15 minutes and needs to be provided in the header of the HTTP request. There are two ways to get the token:
```

File: docs/cloud.md [hunk 2]
```diff
@@ -45,34 +45,34 @@ Generate JWT token :
 JWT_TOKEN=$(ocm token)
 ```
 
-##### HTTP request
+### HTTP request
 
 ```bash
 JWT_TOKEN=$(curl https://sso.redhat.com/auth/realms/redhat-external/protocol/openid-connect/token -d client_id=cloud-services -d grant_type=refresh_token -d refresh_token=${OFFLINE_TOKEN} | jq -r '.access_token')
```
Comment (carbonin) [inline docs/cloud.md:51]: This references the offline token that is described in the `ocm-cli` section which feels like an alternative to this one. Maybe we should move the description of the offline token to the top level `User Authentication` section?

File: docs/cloud.md [hunk 3]
```diff
@@ -45,34 +45,34 @@ Generate JWT token :
...
 ```
 
-The JWT token is valid for 15 minutes and need to be provided in the header of the HTTP request.
+### API Usage example
 
 Here an example to get all the user's clusters:
```
Comment (carbonin) [inline docs/cloud.md:56]:
```suggestion
Here is an example of a request to get all the user's clusters:
```

File: docs/cloud.md [hunk 4]
```diff
@@ -45,34 +45,34 @@ Generate JWT token :
...
 
 ```bash
-curl https://api.openshift.com/api/assisted-install/v1/clusters -H "Authorization: Bearer ${JWT_TOKEN}"
+curl https://api.openshift.com/api/assisted-install/v2/clusters -H "Authorization: Bearer ${JWT_TOKEN}"
 ```
 
-#### Agent Authentication
+## Agent Authentication
 
 Agent authentication uses a token from the Pull Secret and needs to be provided in the header of the HTTP request.
```
Comment (carbonin) [inline docs/cloud.md:64]: Do we want to describe this more specifically?

IIRC the token is the value of a specific key in the pull secret (not the whole thing).

File: docs/cloud.md (hunk 2 above)
Comment (omertuc) [reply docs/cloud.md:51]: yep

File: docs/cloud.md (hunk 1 above)
Comment (omertuc) [reply docs/cloud.md]: right

File: docs/cloud.md (hunk 3 above)
Comment (omertuc) [reply docs/cloud.md]: ok

File: docs/cloud.md (hunk 4 above)
Comment (omertuc) [reply docs/cloud.md:70]: definitely

File: docs/cloud.md [hunk 5]
```diff
@@ -2,37 +2,43 @@
...
+User Authentication is using JWT tokens. The JWT token is valid for 15 minutes and needs to be provided in the header of the HTTP request.
 
-##### ocm client
+First you must obtain your offline token from https://console.redhat.com/openshift/token (This token does not expire) and set it as `OFFLINE_TOKEN`:
+
+```bash
+OFFLINE_TOKEN=<your offline token>
+```
+
+Then there are two to get the JWT token:
```
Comment (mhrivnak) [inline docs/cloud.md:33]: two of what?

File: docs/cloud.md [hunk 6]
```diff
@@ -45,34 +51,36 @@ Generate JWT token :
...
+PULL_SECRET_FILE=$HOME/.docker/config.json
+PULL_SECRET_TOKEN=$(jq -r '.auths."cloud.openshift.com".auth' $PULL_SECRET_FILE)
+curl -X POST https://api.openshift.com/api/assisted-install/v2/clusters/f74fe2e3-1d99-4383-b2f3-8213af03ddeb/hosts -H "X-Secret-Key: <PULL_SECRET_TOKEN>"
  ```
 
- ### UUIDs
+# UUIDs
 
- API objects instances are defined with an unique identifier (UUID).
+API objects instances are defined with an unique identifier (UUID).
```
Comment (mhrivnak) [inline docs/cloud.md:82]: s/objects/object/

File: docs/cloud.md [hunk 7]
```diff
@@ -2,37 +2,43 @@
...
-The API is available here: https://api.openshift.com/api/assisted-install/v1/
+The API is available here: https://api.openshift.com/api/assisted-install/v2/
 
-### Authentication
+## Authentication
 
-On console.redhat.com, Assisted Service APIs calls are authenticated.
+On api.redhat.com, Assisted Service APIs calls are authenticated.
 
 There are two kind of authentications: User and Agent.
```
Comment (mhrivnak) [inline docs/cloud.md:17]: s/two kind of authentications/two kinds of authentication/

**Review by mhrivnak — CHANGES_REQUESTED** [review]
Thanks for updating this doc. I just saw a few more little things.

**Comment by omertuc** [comment]
> Thanks for updating this doc. I just saw a few more little things.

Thanks, fixed

**Review by mhrivnak — APPROVED** [review]

**Pull request merged**